README.md

```diff
@@ -180,6 +180,29 @@ We can always use layer to search for data availability:
   const datesS1 = await layerS1.findDatesUTC(bbox, fromTime, toTime);
 ```
 
+## Getting basic statistics and histogram
+
+Getting basic statistics(mean, min, max, standard deviation) and a histogram for a geometry(Polygon or MultiPolygon).
+The histogram uses the `equalfrequency` binning method and defaults to 5 bins.
+
+```javascript
+  const layerS2L2A = new S2L2ALayer({
+    instanceId: '<my-instance-id>',
+    layerId: 'LAYER_S2L2A',
+    maxCloudCoverPercent: 100,
+  });
+
+const stats = await layerS2L2A.getStatsAndHistogram({
+  geometry: geometry, 
+  crs: CRS_EPSG4326, 
```

Comment:
We are a bit inconsistent here - with `BBox` we specify CRS as part of the structure, while here we are using a separate object. Not sure if it needs to be changed though (and how to best do it), what is your opinion?

Reply:
The underlying FIS request requires a `crs` and the other reason is prefixing the resolution with `m`  when using `4326`, which would be nice to hide from the user in my view. So either we leave it as it is, or require a the crs property in the Geojson Geometry ie

```
type: "Polygon",
coordinates: [...]
crs:{
 type: "name",
 name: "urn:ogc:def:crs:EPSG::4326"
}
```

Reply:
Yes we will need to expand `Polygon` as according to the spec the `crs` member was removed as can be seen at https://tools.ietf.org/html/rfc7946#appendix-B) under section `B.1.  Normative Changes` . Under section `4.  Coordinate Reference System` it is mentioned that all coordinates are in wgs:84/EPSG:4326. So yes we should default to `EPSG:4326`.

Do you still think it is a good idea to expand the `Polygon` declaration to include `crs`?

Reply:
I think it is a good idea to be consistent. :)

The problem here is that this is the interface and if we ever change it, it will be a breaking change, so we need to minimize the chance of this happening, or at least provide a way to upgrade interface in the future.

Since we probably don't want to change `BBox` definition, I would suggest adding `crs` to `Polygon`. If we ever decide that a better type is needed, we can introduce it and keep backwards-compatibility through object inspection. This way the decision shifts to "let's have a single parameter for shape and CRS", which sounds OK to me.

What do you think?

Reply:
I removed `crs` as a param. The crs defaults to `EPSG:4326`, but you can specify the crs in the polygon's geometry with the following
```
type: "Polygon",
coordinates: [...]
crs:{
  type: "name",
  properties: {
    name: "urn:ogc:def:crs:EPSG::4326"
  }
}
```

Using our `Bbox` class also works by passing the `toGeojson()` for the `geometry` params as the method adds `crs` to the `Polygon`

Reply:
The only issue is that `"urn:ogc:def:crs:OGC:1.3:CRS84"` would fail, since we assign `CRS_EPSG4326` to `CRS_WGS84`

Reply:
Changed the function to find the CRS from a `urn` to a switch case, and that resolves the issue where `urn:ogc:def:crs:OGC:1.3:CRS84` would throw an error

Reply:
Nice, looks much better!

Artifact: this line (`crs: ...`) should be removed from README.

Reply:
Thanks, removed the artifact in README and Landsat5 story